akka-projection-core-test/src/test/scala/akka/projection/internal/metrics/tools/InternalProjectionStateMetricsSpec.scala

```diff
@@ -134,25 +134,32 @@ object InternalProjectionStateMetricsSpec {
     val adaptedHandlerStrategy: HandlerStrategy = offsetStrategy match {
       case ExactlyOnce(_) =>
         handlerStrategy match {
-          case SingleHandlerStrategy(handlerFactory) => {
+          case singleHandlerStrategy: SingleHandlerStrategy[Envelope] @unchecked => {
```

Comment:
Scala3 complained here, so this was the only way I could make it compile, same below.